### PR TITLE
 Integrate pending PR's to be merged plus adding of tests for them inside this filter

### PR DESCRIPTION
**Pull request by purbon**
PR that will basically integrate many of the pending Pr's to be merged, including a new test suite for the issues down there.

### Changes for 2.1.0
  - Improved the configuration options to be more easy to understand and
    match what the expectations are from the documentation.
  - Initial refactoring to include later one a common client for all the
    ES plugins.
  - Adding support for having an index in the query pattern.
  - Improved documentation.
  - Added initial integration and unit tests.

more changes might be added in feature rounds of maintenance.

Fixes #17 #14

### CODE REVIEW

**Comment by purbon** [comment]
@jsvd your thoughts are welcome here.

**Comment by purbon** [comment]
not sure why test are red in travis, will need to investigate.

File: Gemfile [hunk 1]
```diff
@@ -1,2 +1,3 @@
 source 'https://rubygems.org'
-gemspec
\ No newline at end of file
+gemspec
+gem "pry", :group => :development
```
Comment (untergeek) [inline Gemfile]: Did you add this for testing and intend for it to stay after the tests?

File: spec/filters/elasticsearch_spec.rb [hunk 2]
```diff
@@ -15,4 +15,69 @@
...
+        }
+      end
+
+      let(:response) do
+        LogStash::Json.load(File.read(File.join(File.dirname(__FILE__), "fixtures", "request_x_10.json")))
+      end
+
+      it "should enhance the current event with new data" do
+        plugin.filter(event)
+        expect(event["code"]).to eq([404]*10)
```
Comment (untergeek) [inline spec/filters/elasticsearch_spec.rb]: This is cool.  I didn't know you could do that `*10` to catch 10 404 events.

**Comment by untergeek** [comment]
LGTM™, other than wondering whether we want to keep `pry` as a development dependency, since that's the only place it exists, and it's interactive by nature.

**Comment by suyograo** [comment]
Keep in mind this needs to be backported to plugin-api-v1

File: Gemfile (hunk 1 above)
Comment (purbon) [inline Gemfile]: yes, added this for testing. I like that we have it somehow here, but is just my taste for debugging :-P

Comment (purbon) [inline Gemfile]: this is added here also as development as you commented out.

**Comment by purbon** [comment]
@suyograo yes, will backport to previous version, thanks for the reminder.

**Pull request closed without merging**